sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -174,20 +175,38 @@ def get_final_get_url(self, pipeline_response):
         """
         raise NotImplementedError()
 
+class _LroOption(str, Enum):
+    """Known LRO options from Swagger."""
+
+    FINAL_STATE_VIA = "final-state-via"
+
+
+class _FinalStateViaOption(str, Enum):
+    """Possible final-state-via options."""
+
+    AZURE_ASYNC_OPERATION_FINAL_STATE = "azure-async-operation"
+    LOCATION_FINAL_STATE = "location"
+    OPERATION_LOCATION_FINAL_STATE = "operation-location"
```

Comment:
Are these case-sensitive?

Reply:
not case sensitive. These are defined header names where our pollers can get the final get url. It's private too, so naming doesn't matter too much

Reply:
If they are not case sensitive, shouldn't we use CaseInsensitiveEnumMeta?

Reply:
From the doc https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation-options 

the value of final-state-via is string - one of azure-async-operation or location or original-uri or operation-location.

Do you mean we don't need special handler for original-uri?

Reply:
In terms of using CaseInsensitiveEnumMeta, I mean the value is not case sensitive, it doesn't really matter if the enum name is case sensitive or not since we're the only one using it. So I don't think we need to overcomplicate the private enum by adding CaseInsensitiveEnumMeta as its metaclass

Reply:
Yeah, right now I don't think we're using a special handler for original url. We may eventually add support for this (cc @lmazuel since I'm not sure), but for now we're not doing any explicit handling of it, and it's outside of the scope of this PR

Reply:
Values are from autorest core, this is not a wire format that service teams can mess up, so we know it will never be in different casing. I would keep it simple with lower case and not meta enum